build failure: multiple definition of `pthread_create'

c-blosc fails to build on Windows with gcc with the message `multiple definition of 'pthread_create'`.

The issue seems to be with the file `c-blosc/blosc/blosc.c` and the following definition block:

```
#if defined(_WIN32)
  #include "win32/pthread.h"
  #include "win32/pthread.c"
#else
  #include <pthread.h>
#endif
```
It seem the win32 gcc already provides the functions that are provided by `win32/pthread.c` leading to conflicting definitions.

On my machine testing for `__GNUC__` solves this issue:
```
#if defined(_WIN32) && !defined(__GNUC__)
  #include "win32/pthread.h"
  #include "win32/pthread.c"
#else
  #include <pthread.h>
#endif
```
I am not sure however, if this breaks things in other win32 configurations.

```
C:\bcolz>gcc --version
gcc (tdm-1) 4.9.2
```

Build log:
```
[bquerymp] C:\bcolz>python setup.py build_ext --inplace
* Found Cython 0.22 package installed.
* Found numpy 1.9.1 package installed.
* Found numexpr 2.3.1 package installed.
running build_ext
cythoning bcolz/carray_ext.pyx to bcolz\carray_ext.c
building 'bcolz.carray_ext' extension
C compiler: gcc -O2 -Wall -Wstrict-prototypes

compile options: '-DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D__MSVCRT_VERSION_
_=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\numpy\core\includ
e -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc/internal-compli
bs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anaconda\envs\bquery
mp\include -IC:\Anaconda\envs\bquerymp\PC -c'
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/blosc\shuff
le.c -o build\temp.win32-2.7\Release\c-blosc\blosc\shuffle.o
Found executable C:\Program Files\TDM-GCC-32\bin\gcc.exe
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\deflate.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\deflate.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\gzread.c -o build\temp.win32-2.7\Release\c-blosc\internal-comp
libs\zlib-1.2.8\gzread.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\lz4-r119\lz4hc.c -o build\temp.win32-2.7\Release\c-blosc\internal-complib
s\lz4-r119\lz4hc.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\zutil.c -o build\temp.win32-2.7\Release\c-blosc\internal-compl
ibs\zlib-1.2.8\zutil.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/blosc\blosc
lz.c -o build\temp.win32-2.7\Release\c-blosc\blosc\blosclz.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\compress.c -o build\temp.win32-2.7\Release\c-blosc\internal-co
mplibs\zlib-1.2.8\compress.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c bcolz\carray_ext.c
-o build\temp.win32-2.7\Release\bcolz\carray_ext.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\lz4-r119\lz4.c -o build\temp.win32-2.7\Release\c-blosc\internal-complibs\
lz4-r119\lz4.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\inflate.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\inflate.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\trees.c -o build\temp.win32-2.7\Release\c-blosc\internal-compl
ibs\zlib-1.2.8\trees.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\inffast.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\inffast.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\snappy-1.1.1\snappy-c.cc -o build\temp.win32-2.7\Release\c-blosc\internal
-complibs\snappy-1.1.1\snappy-c.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\uncompr.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\uncompr.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\gzwrite.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\gzwrite.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\snappy-1.1.1\snappy.cc -o build\temp.win32-2.7\Release\c-blosc\internal-c
omplibs\snappy-1.1.1\snappy.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\infback.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\infback.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\inftrees.c -o build\temp.win32-2.7\Release\c-blosc\internal-co
mplibs\zlib-1.2.8\inftrees.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\gzclose.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\gzclose.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\gzlib.c -o build\temp.win32-2.7\Release\c-blosc\internal-compl
ibs\zlib-1.2.8\gzlib.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\crc32.c -o build\temp.win32-2.7\Release\c-blosc\internal-compl
ibs\zlib-1.2.8\crc32.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\snappy-1.1.1\snappy-sinksource.cc -o build\temp.win32-2.7\Release\c-blosc
\internal-complibs\snappy-1.1.1\snappy-sinksource.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\snappy-1.1.1\snappy-stubs-internal.cc -o build\temp.win32-2.7\Release\c-b
losc\internal-complibs\snappy-1.1.1\snappy-stubs-internal.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/blosc\blosc
.c -o build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o
gcc -O2 -Wall -Wstrict-prototypes -DHAVE_LZ4=1 -DHAVE_SNAPPY=1 -DHAVE_ZLIB=1 -D_
_MSVCRT_VERSION__=0x0900 -Ibcolz -IC:\Anaconda\envs\bquerymp\lib\site-packages\n
umpy\core\include -Ic-blosc\blosc -Ic-blosc/internal-complibs\lz4-r119 -Ic-blosc
/internal-complibs\snappy-1.1.1 -Ic-blosc/internal-complibs\zlib-1.2.8 -IC:\Anac
onda\envs\bquerymp\include -IC:\Anaconda\envs\bquerymp\PC -c c-blosc/internal-co
mplibs\zlib-1.2.8\adler32.c -o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\adler32.o
g++ -shared build\temp.win32-2.7\Release\bcolz\carray_ext.o build\temp.win32-2.7
\Release\c-blosc\blosc\blosc.o build\temp.win32-2.7\Release\c-blosc\blosc\bloscl
z.o build\temp.win32-2.7\Release\c-blosc\blosc\shuffle.o build\temp.win32-2.7\Re
lease\c-blosc\internal-complibs\lz4-r119\lz4.o build\temp.win32-2.7\Release\c-bl
osc\internal-complibs\lz4-r119\lz4hc.o build\temp.win32-2.7\Release\c-blosc\inte
rnal-complibs\snappy-1.1.1\snappy-c.o build\temp.win32-2.7\Release\c-blosc\inter
nal-complibs\snappy-1.1.1\snappy-sinksource.o build\temp.win32-2.7\Release\c-blo
sc\internal-complibs\snappy-1.1.1\snappy-stubs-internal.o build\temp.win32-2.7\R
elease\c-blosc\internal-complibs\snappy-1.1.1\snappy.o build\temp.win32-2.7\Rele
ase\c-blosc\internal-complibs\zlib-1.2.8\adler32.o build\temp.win32-2.7\Release\
c-blosc\internal-complibs\zlib-1.2.8\compress.o build\temp.win32-2.7\Release\c-b
losc\internal-complibs\zlib-1.2.8\crc32.o build\temp.win32-2.7\Release\c-blosc\i
nternal-complibs\zlib-1.2.8\deflate.o build\temp.win32-2.7\Release\c-blosc\inter
nal-complibs\zlib-1.2.8\gzclose.o build\temp.win32-2.7\Release\c-blosc\internal-
complibs\zlib-1.2.8\gzlib.o build\temp.win32-2.7\Release\c-blosc\internal-compli
bs\zlib-1.2.8\gzread.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zl
ib-1.2.8\gzwrite.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1
.2.8\infback.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8
\inffast.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\inf
late.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\inftree
s.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\trees.o bu
ild\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\uncompr.o build\
temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\zutil.o -LC:\Anacond
a\envs\bquerymp\libs -LC:\Anaconda\envs\bquerymp\PCbuild -lpython27 -lmsvcr90 -o
 C:\bcolz\bcolz\carray_ext.pyd
Found executable C:\Program Files\TDM-GCC-32\bin\g++.exe
C:/Program Files/TDM-GCC-32/bin/../lib/gcc/mingw32/4.9.2/../../../libpthread.a(l
ibwinpthread_la-thread.o):thread.c:(.text+0x3710): multiple definition of `pthre
ad_create'
build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o:blosc.c:(.text+0x180): first
defined here
C:/Program Files/TDM-GCC-32/bin/../lib/gcc/mingw32/4.9.2/../../../libpthread.a(l
ibwinpthread_la-cond.o):cond.c:(.text+0x3d0): multiple definition of `pthread_co
nd_init'
build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o:blosc.c:(.text+0x260): first
defined here
C:/Program Files/TDM-GCC-32/bin/../lib/gcc/mingw32/4.9.2/../../../libpthread.a(l
ibwinpthread_la-cond.o):cond.c:(.text+0x980): multiple definition of `pthread_co
nd_destroy'
build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o:blosc.c:(.text+0x300): first
defined here
C:/Program Files/TDM-GCC-32/bin/../lib/gcc/mingw32/4.9.2/../../../libpthread.a(l
ibwinpthread_la-cond.o):cond.c:(.text+0xb80): multiple definition of `pthread_co
nd_signal'
build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o:blosc.c:(.text+0x3e0): first
defined here
C:/Program Files/TDM-GCC-32/bin/../lib/gcc/mingw32/4.9.2/../../../libpthread.a(l
ibwinpthread_la-cond.o):cond.c:(.text+0xca0): multiple definition of `pthread_co
nd_broadcast'
build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o:blosc.c:(.text+0x440): first
defined here
C:/Program Files/TDM-GCC-32/bin/../lib/gcc/mingw32/4.9.2/../../../libpthread.a(l
ibwinpthread_la-cond.o):cond.c:(.text+0xdb0): multiple definition of `pthread_co
nd_wait'
build\temp.win32-2.7\Release\c-blosc\blosc\blosc.o:blosc.c:(.text+0x340): first
defined here
collect2.exe: error: ld returned 1 exit status
error: Command "g++ -shared build\temp.win32-2.7\Release\bcolz\carray_ext.o buil
d\temp.win32-2.7\Release\c-blosc\blosc\blosc.o build\temp.win32-2.7\Release\c-bl
osc\blosc\blosclz.o build\temp.win32-2.7\Release\c-blosc\blosc\shuffle.o build\t
emp.win32-2.7\Release\c-blosc\internal-complibs\lz4-r119\lz4.o build\temp.win32-
2.7\Release\c-blosc\internal-complibs\lz4-r119\lz4hc.o build\temp.win32-2.7\Rele
ase\c-blosc\internal-complibs\snappy-1.1.1\snappy-c.o build\temp.win32-2.7\Relea
se\c-blosc\internal-complibs\snappy-1.1.1\snappy-sinksource.o build\temp.win32-2
.7\Release\c-blosc\internal-complibs\snappy-1.1.1\snappy-stubs-internal.o build\
temp.win32-2.7\Release\c-blosc\internal-complibs\snappy-1.1.1\snappy.o build\tem
p.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\adler32.o build\temp.wi
n32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\compress.o build\temp.win32
-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\crc32.o build\temp.win32-2.7\R
elease\c-blosc\internal-complibs\zlib-1.2.8\deflate.o build\temp.win32-2.7\Relea
se\c-blosc\internal-complibs\zlib-1.2.8\gzclose.o build\temp.win32-2.7\Release\c
-blosc\internal-complibs\zlib-1.2.8\gzlib.o build\temp.win32-2.7\Release\c-blosc
\internal-complibs\zlib-1.2.8\gzread.o build\temp.win32-2.7\Release\c-blosc\inte
rnal-complibs\zlib-1.2.8\gzwrite.o build\temp.win32-2.7\Release\c-blosc\internal
-complibs\zlib-1.2.8\infback.o build\temp.win32-2.7\Release\c-blosc\internal-com
plibs\zlib-1.2.8\inffast.o build\temp.win32-2.7\Release\c-blosc\internal-complib
s\zlib-1.2.8\inflate.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zl
ib-1.2.8\inftrees.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-
1.2.8\trees.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\
uncompr.o build\temp.win32-2.7\Release\c-blosc\internal-complibs\zlib-1.2.8\zuti
l.o -LC:\Anaconda\envs\bquerymp\libs -LC:\Anaconda\envs\bquerymp\PCbuild -lpytho
n27 -lmsvcr90 -o C:\bcolz\bcolz\carray_ext.pyd" failed with exit status 1
```

--- a/blosc/CMakeLists.txt
+++ b/blosc/CMakeLists.txt
@@ -98,7 +98,7 @@ set_target_properties(blosc_shared PROPERTIES
         SOVERSION 1  # Change this when an ABI change happens
     )
 if (MSVC)
-	set_target_properties (blosc_shared PROPERTIES COMPILE_FLAGS -DDLL_EXPORT=TRUE)
+	set_target_properties (blosc_shared PROPERTIES COMPILE_FLAGS -DBLOSC_DLL_EXPORT=TRUE)
 endif(MSVC)
 
 if(COMPILER_SUPPORT_SSE2)

--- a/blosc/blosc.c
+++ b/blosc/blosc.c
@@ -49,7 +49,7 @@
   #include <inttypes.h>
 #endif  /* _WIN32 */
 
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(__GNUC__)
   #include "win32/pthread.h"
   #include "win32/pthread.c"
 #else

--- a/blosc/blosc.h
+++ b/blosc/blosc.h
@@ -14,12 +14,12 @@
 extern "C" {
 #endif
 
-#ifdef DLL_EXPORT
-    #undef DLL_EXPORT
-	#define DLL_EXPORT __declspec(dllexport)
+#ifdef BLOSC_DLL_EXPORT
+    #undef BLOSC_DLL_EXPORT
+	#define BLOSC_DLL_EXPORT __declspec(dllexport)
 #else
-    #undef DLL_EXPORT
-	#define DLL_EXPORT
+    #undef BLOSC_DLL_EXPORT
+	#define BLOSC_DLL_EXPORT
 #endif
 
 /* Version numbers */
@@ -109,7 +109,7 @@ extern "C" {
   which case you should *exclusively* use the
   blosc_compress_ctx()/blosc_decompress_ctx() pair (see below).
   */
-DLL_EXPORT void blosc_init(void);
+BLOSC_DLL_EXPORT void blosc_init(void);
 
 
 /**
@@ -119,7 +119,7 @@ DLL_EXPORT void blosc_init(void);
   unless you have not used blosc_init() before (see blosc_init()
   above).
   */
-DLL_EXPORT void blosc_destroy(void);
+BLOSC_DLL_EXPORT void blosc_destroy(void);
 
 
 /**
@@ -156,9 +156,9 @@ DLL_EXPORT void blosc_destroy(void);
   should never happen.  If you see this, please report it back
   together with the buffer data causing this and compression settings.
   */
-DLL_EXPORT int blosc_compress(int clevel, int doshuffle, size_t typesize,
-                              size_t nbytes, const void *src, void *dest,
-                              size_t destsize);
+BLOSC_DLL_EXPORT int blosc_compress(int clevel, int doshuffle, size_t typesize,
+                                    size_t nbytes, const void *src, void *dest,
+                                    size_t destsize);
 
 
 /**
@@ -180,10 +180,10 @@ DLL_EXPORT int blosc_compress(int clevel, int doshuffle, size_t typesize,
   should never happen.  If you see this, please report it back
   together with the buffer data causing this and compression settings.
 */
-DLL_EXPORT int blosc_compress_ctx(int clevel, int doshuffle, size_t typesize,
-                                  size_t nbytes, const void* src, void* dest,
-                                  size_t destsize, const char* compressor,
-                                  size_t blocksize, int numinternalthreads);
+BLOSC_DLL_EXPORT int blosc_compress_ctx(int clevel, int doshuffle, size_t typesize,
+                                        size_t nbytes, const void* src, void* dest,
+                                        size_t destsize, const char* compressor,
+                                        size_t blocksize, int numinternalthreads);
 
 /**
   Decompress a block of compressed data in `src`, put the result in
@@ -198,7 +198,7 @@ DLL_EXPORT int blosc_compress_ctx(int clevel, int doshuffle, size_t typesize,
   output buffer is not large enough, then 0 (zero) or a negative value
   will be returned instead.
 */
-DLL_EXPORT int blosc_decompress(const void *src, void *dest, size_t destsize);
+BLOSC_DLL_EXPORT int blosc_decompress(const void *src, void *dest, size_t destsize);
 
 
 /**
@@ -218,7 +218,7 @@ DLL_EXPORT int blosc_decompress(const void *src, void *dest, size_t destsize);
   output buffer is not large enough, then 0 (zero) or a negative value
   will be returned instead.
 */
-DLL_EXPORT int blosc_decompress_ctx(const void *src, void *dest,
+BLOSC_DLL_EXPORT int blosc_decompress_ctx(const void *src, void *dest,
                                     size_t destsize, int numinternalthreads);
 
 /**
@@ -229,7 +229,7 @@ DLL_EXPORT int blosc_decompress_ctx(const void *src, void *dest,
   Returns the number of bytes copied to `dest` or a negative value if
   some error happens.
   */
-DLL_EXPORT int blosc_getitem(const void *src, int start, int nitems, void *dest);
+BLOSC_DLL_EXPORT int blosc_getitem(const void *src, int start, int nitems, void *dest);
 
 
 /**
@@ -240,7 +240,7 @@ DLL_EXPORT int blosc_getitem(const void *src, int start, int nitems, void *dest)
 
   Returns the previous number of threads.
   */
-DLL_EXPORT int blosc_set_nthreads(int nthreads);
+BLOSC_DLL_EXPORT int blosc_set_nthreads(int nthreads);
 
 
 /**
@@ -252,7 +252,7 @@ DLL_EXPORT int blosc_set_nthreads(int nthreads);
   for it in this build, it returns a -1.  Else it returns the code for
   the compressor (>=0).
   */
-DLL_EXPORT int blosc_set_compressor(const char* compname);
+BLOSC_DLL_EXPORT int blosc_set_compressor(const char* compname);
 
 
 /**
@@ -262,7 +262,7 @@ DLL_EXPORT int blosc_set_compressor(const char* compname);
   for it in this build, -1 is returned.  Else, the compressor code is
   returned.
  */
-DLL_EXPORT int blosc_compcode_to_compname(int compcode, char **compname);
+BLOSC_DLL_EXPORT int blosc_compcode_to_compname(int compcode, char **compname);
 
 
 /**
@@ -271,7 +271,7 @@ DLL_EXPORT int blosc_compcode_to_compname(int compcode, char **compname);
   If the compressor name is not recognized, or there is not support
   for it in this build, -1 is returned instead.
  */
-DLL_EXPORT int blosc_compname_to_compcode(const char *compname);
+BLOSC_DLL_EXPORT int blosc_compname_to_compcode(const char *compname);
 
 
 /**
@@ -285,7 +285,7 @@ DLL_EXPORT int blosc_compname_to_compcode(const char *compname);
 
   This function should always succeed.
   */
-DLL_EXPORT char* blosc_list_compressors(void);
+BLOSC_DLL_EXPORT char* blosc_list_compressors(void);
 
 
 /**
@@ -302,7 +302,7 @@ DLL_EXPORT char* blosc_list_compressors(void);
   If the compressor is supported, it returns the code for the library
   (>=0).  If it is not supported, this function returns -1.
   */
-DLL_EXPORT int blosc_get_complib_info(char *compname, char **complib, char **version);
+BLOSC_DLL_EXPORT int blosc_get_complib_info(char *compname, char **complib, char **version);
 
 
 /**
@@ -311,7 +311,7 @@ DLL_EXPORT int blosc_get_complib_info(char *compname, char **complib, char **ver
   problems releasing the resources, it returns a negative number, else
   it returns 0.
   */
-DLL_EXPORT int blosc_free_resources(void);
+BLOSC_DLL_EXPORT int blosc_free_resources(void);
 
 
 /**
@@ -325,8 +325,8 @@ DLL_EXPORT int blosc_free_resources(void);
 
   This function should always succeed.
   */
-DLL_EXPORT void blosc_cbuffer_sizes(const void *cbuffer, size_t *nbytes,
-                         size_t *cbytes, size_t *blocksize);
+BLOSC_DLL_EXPORT void blosc_cbuffer_sizes(const void *cbuffer, size_t *nbytes,
+                                          size_t *cbytes, size_t *blocksize);
 
 
 /**
@@ -343,8 +343,8 @@ DLL_EXPORT void blosc_cbuffer_sizes(const void *cbuffer, size_t *nbytes,
 
   This function should always succeed.
   */
-DLL_EXPORT void blosc_cbuffer_metainfo(const void *cbuffer, size_t *typesize,
-                            int *flags);
+BLOSC_DLL_EXPORT void blosc_cbuffer_metainfo(const void *cbuffer, size_t *typesize,
+                                             int *flags);
 
 
 /**
@@ -354,8 +354,8 @@ DLL_EXPORT void blosc_cbuffer_metainfo(const void *cbuffer, size_t *typesize,
 
   This function should always succeed.
   */
-DLL_EXPORT void blosc_cbuffer_versions(const void *cbuffer, int *version,
-                            int *versionlz);
+BLOSC_DLL_EXPORT void blosc_cbuffer_versions(const void *cbuffer, int *version,
+                                             int *versionlz);
 
 
 /**
@@ -363,7 +363,7 @@ DLL_EXPORT void blosc_cbuffer_versions(const void *cbuffer, int *version,
 
   This function should always succeed.
   */
-DLL_EXPORT char *blosc_cbuffer_complib(const void *cbuffer);
+BLOSC_DLL_EXPORT char *blosc_cbuffer_complib(const void *cbuffer);
 
 
 
@@ -378,7 +378,7 @@ DLL_EXPORT char *blosc_cbuffer_complib(const void *cbuffer);
   Force the use of a specific blocksize.  If 0, an automatic
   blocksize will be used (the default).
   */
-DLL_EXPORT void blosc_set_blocksize(size_t blocksize);
+BLOSC_DLL_EXPORT void blosc_set_blocksize(size_t blocksize);
 
 #ifdef __cplusplus
 }

--- a/blosc/config.h.in
+++ b/blosc/config.h.in
@@ -4,7 +4,7 @@
 #cmakedefine HAVE_LZ4 @HAVE_LZ4@
 #cmakedefine HAVE_SNAPPY @HAVE_SNAPPY@
 #cmakedefine HAVE_ZLIB @HAVE_ZLIB@
-#cmakedefine DLL_EXPORT @DLL_EXPORT@
+#cmakedefine BLOSC_DLL_EXPORT @DLL_EXPORT@
 
 
 #endif

--- a/blosc/shuffle-avx2.h
+++ b/blosc/shuffle-avx2.h
@@ -20,14 +20,14 @@ extern "C" {
 /**
   AVX2-accelerated shuffle routine.
 */
-DLL_EXPORT void shuffle_avx2(const size_t bytesoftype, const size_t blocksize,
-             const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void shuffle_avx2(const size_t bytesoftype, const size_t blocksize,
+                                   const uint8_t* const _src, uint8_t* const _dest);
 
 /**
   AVX2-accelerated unshuffle routine.
 */
-DLL_EXPORT void unshuffle_avx2(const size_t bytesoftype, const size_t blocksize,
-               const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void unshuffle_avx2(const size_t bytesoftype, const size_t blocksize,
+                                     const uint8_t* const _src, uint8_t* const _dest);
 
 #ifdef __cplusplus
 }

--- a/blosc/shuffle-common.h
+++ b/blosc/shuffle-common.h
@@ -10,12 +10,12 @@
 #define SHUFFLE_COMMON_H
 
 /* Macro for specifying an exported function. */
-#if defined(_WIN32) && defined(DLL_EXPORT)
-  #undef DLL_EXPORT
-  #define DLL_EXPORT __declspec(dllexport)
+#if defined(_WIN32) && defined(BLOSC_DLL_EXPORT)
+  #undef BLOSC_DLL_EXPORT
+  #define BLOSC_DLL_EXPORT __declspec(dllexport)
 #else
-  #undef DLL_EXPORT
-  #define DLL_EXPORT
+  #undef BLOSC_DLL_EXPORT
+  #define BLOSC_DLL_EXPORT
 #endif
 
 /* Define the __SSE2__ symbol if compiling with Visual C++ and

--- a/blosc/shuffle-generic.h
+++ b/blosc/shuffle-generic.h
@@ -24,14 +24,14 @@ extern "C" {
 /**
   Generic (non-hardware-accelerated) shuffle routine.
 */
-DLL_EXPORT void shuffle_generic(const size_t bytesoftype, const size_t blocksize,
-             const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void shuffle_generic(const size_t bytesoftype, const size_t blocksize,
+                                      const uint8_t* const _src, uint8_t* const _dest);
 
 /**
   Generic (non-hardware-accelerated) unshuffle routine.
 */
-DLL_EXPORT void unshuffle_generic(const size_t bytesoftype, const size_t blocksize,
-               const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void unshuffle_generic(const size_t bytesoftype, const size_t blocksize,
+                                        const uint8_t* const _src, uint8_t* const _dest);
 
 #ifdef __cplusplus
 }

--- a/blosc/shuffle-sse2.h
+++ b/blosc/shuffle-sse2.h
@@ -20,14 +20,14 @@ extern "C" {
 /**
   SSE2-accelerated shuffle routine.
 */
-DLL_EXPORT void shuffle_sse2(const size_t bytesoftype, const size_t blocksize,
-             const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void shuffle_sse2(const size_t bytesoftype, const size_t blocksize,
+                                   const uint8_t* const _src, uint8_t* const _dest);
 
 /**
   SSE2-accelerated unshuffle routine.
 */
-DLL_EXPORT void unshuffle_sse2(const size_t bytesoftype, const size_t blocksize,
-               const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void unshuffle_sse2(const size_t bytesoftype, const size_t blocksize,
+                                     const uint8_t* const _src, uint8_t* const _dest);
 
 #ifdef __cplusplus
 }

--- a/blosc/shuffle.h
+++ b/blosc/shuffle.h
@@ -31,8 +31,8 @@ extern "C" {
   calling the hardware-accelerated routines because this method is both cross-
   platform and future-proof.
 */
-DLL_EXPORT void shuffle(const size_t bytesoftype, const size_t blocksize,
-             const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void shuffle(const size_t bytesoftype, const size_t blocksize,
+                              const uint8_t* const _src, uint8_t* const _dest);
 
 /**
   Primary unshuffle routine.
@@ -44,8 +44,8 @@ DLL_EXPORT void shuffle(const size_t bytesoftype, const size_t blocksize,
   calling the hardware-accelerated routines because this method is both cross-
   platform and future-proof.
 */
-DLL_EXPORT void unshuffle(const size_t bytesoftype, const size_t blocksize,
-               const uint8_t* const _src, uint8_t* const _dest);
+BLOSC_DLL_EXPORT void unshuffle(const size_t bytesoftype, const size_t blocksize,
+                                const uint8_t* const _src, uint8_t* const _dest);
 
 #ifdef __cplusplus
 }

